addons/mail/static/src/new/web/messaging_menu/messaging_menu.js

```diff
@@ -36,6 +36,29 @@ export class MessagingMenu extends Component {
         });
     }
 
+    beforeOpen() {
+        this.messaging.fetchPreviews();
+        if (
+            !this.store.discuss.inbox.isLoaded &&
+            this.store.discuss.inbox.status !== "loading" &&
+            this.store.discuss.inbox.counter !== this.store.discuss.inbox.messageIds.length
+        ) {
+            this.threadService.fetchMessages(this.store.discuss.inbox);
+        }
+    }
+
+    onClickPreview(isMarkAsRead, thread) {
+        if (!isMarkAsRead) {
+            this.openDiscussion(thread);
+            return;
+        }
+        if (thread.hasNeedactionMessages) {
+            this.threadService.markAllMessagesAsRead(thread);
+        } else {
+            this.threadService.markAsRead(thread);
```

Comment:
I still think this part is not correct, last seen is handled from the thread view itself, not merely clicking on preview.